### PR TITLE
Watchlist accessibility improvements

### PR DESCRIPTION
**Pull request by mazevedofs**
**Phabricator:** https://phabricator.wikimedia.org/T342359

### Notes
* This PR adds some accessibility adjustments for voiceOver users to make the watchlist feature easier to navigate
* The main difference is that when VoiceOver is running, tapping on the cell doesn't take you directly to the diff. Instead, it opens the menu with "Go to Diff" as the first action. That was done to make it less confusing to VoiceOver users and to concentrate all actions in one.

### Test Steps
1. With VoiceOver on, access the Watchlist Feature from Article. Add an article to the watchlist, choose the expiration date, and ensure the toast is accessible.
2. Access the watchlist from Account in Settings, make sure it's easy to navigate.
3. Try it on a fresh install so the onboarding modal shows

### CODE REVIEW

File: Components/Sources/Components/Components/Watchlist/WKWatchlistView.swift [hunk 1]
```diff
@@ -3,84 +3,95 @@ import WKData
...
+            if viewModel.sections.count > 0 {
+                WKWatchlistContentView(viewModel: viewModel, delegate: delegate, menuButtonDelegate: menuButtonDelegate)
+            } else if viewModel.sections.count == 0 && viewModel.activeFilterCount > 0 {
+                WKEmptyView(viewModel: emptyViewModel, delegate: emptyViewDelegate, type: .filter)
+            } else {
+                WKEmptyView(viewModel: emptyViewModel, delegate: emptyViewDelegate, type: .noItems)
+            }
+        } else {
+            ProgressView()
+            .accessibilityHidden(true)
```
Comment (staykids) [inline Components/Sources/Components/Components/Watchlist/WKWatchlistView.swift:42]: Should we retain the intrinsic accessibility nature of this view here? Using the modifier means the user gets trapped with no indication about the view itself as it's loading, which could be especially problematic if it's taking a long time.

File: Components/Sources/Components/Components/Shared/Buttons/WKMenuButtonDelegate.swift [hunk 2]
```diff
@@ -5,10 +5,12 @@ public protocol WKMenuButtonDelegate: AnyObject {
...
 
 	func wkSwiftUIMenuButtonUserDidTap(configuration: WKMenuButton.Configuration, item: WKMenuButton.MenuItem?)
+    func wkSwiftUIMenuButtonUserDidTapAccessibility(configuration: WKMenuButton.Configuration, item: WKMenuButton.MenuItem?)
 }
 
 public extension WKMenuButtonDelegate {
 	func wkMenuButton(_ sender: WKMenuButton, didTapMenuItem item: WKMenuButton.MenuItem) {}
 	func wkMenuButtonDidTap(_ sender: WKMenuButton) {}
 	func wkSwiftUIMenuButtonUserDidTap(configuration: WKMenuButton.Configuration, item: WKMenuButton.MenuItem?) {}
+    func wkSwiftUIMenuButtonUserDidTapAccessibility(configuration: WKMenuButton.Configuration, item: WKMenuButton.MenuItem?) {}
```
Comment (staykids) [inline Components/Sources/Components/Components/Shared/Buttons/WKMenuButtonDelegate.swift:15]: Same note as elsewhere about finding a way to remove the need for this extra delegate method here and above.

**Review by staykids — CHANGES_REQUESTED** [review]
Creative use of the menu button for accessibility here! I think there are several changes we need to make here and conflicts with main to resolve.

File: Components/Sources/Components/Components/Watchlist/WKWatchlistView.swift [hunk 3]
```diff
@@ -3,84 +3,95 @@ import WKData
...
+            }
+            .padding([.leading, .trailing], 16)
+            .padding([.top], 12)
+        }
+    }
+
+    func getAccessibilityLabelForItem(item: WKWatchlistViewModel.ItemViewModel) -> String {
+        var accessibilityString = String()
+        if let project = viewModel.localizedStrings.localizedProjectNames[item.project] {
+            accessibilityString = "\(project) \(item.timestampStringAccessibility). \(item.title). \(viewModel.localizedStrings.userAccessibility): \(item.username). \(viewModel.localizedStrings.summaryAccessibility): \(item.comment)"
```
[the remark this answers is not in the document]
Comment (mazevedofs) [reply Components/Sources/Components/Components/Watchlist/WKWatchlistView.swift:91]: I'm working on this one and will push it soon. All the other minor ones were addressed.

File: Components/Sources/Components/Components/Watchlist/WKWatchlistViewController.swift [hunk 4]
```diff
@@ -37,46 +37,80 @@ public final class WKWatchlistViewController: WKCanvasViewController {
...
+            } else if tappedTitle == menuButtonItems[2].title {
+                watchlistDelegate?.watchlistUserDidTapUser(project: wkProject, title: title, revisionID: revisionID, oldRevisionID: oldRevisionId, username: username, action: .userContributions)
+                watchlistLoggingDelegate?.logWatchlistUserDidTapUserButtonAction(project: wkProject, action: .userContributions)
+            } else if tappedTitle == menuButtonItems[3].title {
+                watchlistDelegate?.watchlistUserDidTapUser(project: wkProject, title: title, revisionID: revisionID, oldRevisionID: oldRevisionId, username: username, action: .thank(revisionID: revisionID))
+                watchlistLoggingDelegate?.logWatchlistUserDidTapUserButtonAction(project: wkProject, action: .thank(revisionID: revisionID))
+            }
+		}
+
+        func wkSwiftUIMenuButtonUserDidTapAccessibility(configuration: WKMenuButton.Configuration, item: WKMenuButton.MenuItem?) {
```
[the remark this answers is not in the document]
Comment (staykids) [reply Components/Sources/Components/Components/Watchlist/WKWatchlistViewController.swift:86]: Made Phab task for this: https://phabricator.wikimedia.org/T347433

**Review by staykids — APPROVED** [review]

**Pull request merged**